server/src/main/java/org/elasticsearch/rest/CompatibleVersion.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.xcontent.ParsedMediaType;
+
+/**
+ * An interface used to specify a function that returns a compatible API version
+ * Intended to be used in a code base instead of a plugin.
```

Comment:
not sure i follow the second sentence (Intended to be ...)

Reply:
right.. I rephrased this, but maybe it is still to vague. let me know
My intention was to not describe the actual plugin implementation, but just to mention that this abstract the way we actually provide the `logic`.
It happens that we use a plugin for this, but could if we want to use any class within a server to do this as well (we did in the previous attempts)